debezium-testing/debezium-testing-testcontainers/src/main/java/io/debezium/testing/testcontainers/PostgresInfrastructure.java

```diff
@@ -47,10 +48,28 @@ private PostgresInfrastructure(String postgresImageName) {
                 .withNetworkAliases("postgres");
     }
 
-    public static PostgresInfrastructure getDebeziumPostgresInfrastructure() {
+    private PostgresInfrastructure(String postgresImageName, Network network) {
```

Comment:
The network doesn't have to be injected. The Infrastructure classes should imo just return a blueprint for a container.
Tests that use that containers can still configure the network with the `withNetwork()` method.
It think we can remove this method and the new getters that come with the network parameter.

Reply:
Yes agreed. It can passed with the container itself.